### PR TITLE
Fix #280: Add searchType: frontmatter.exists for tracking field existence

### LINKED ISSUE #280: Tracking days when frontmatter field is non empty
I have a field in the frontmatter which could have any text value. I want to track every day in which said field is not empty. How is is possible to achieve this?

### PR DESCRIPTION
**Pull request by mikegreiner**
# Description
Fixes #280 - Adds new `searchType: frontmatter.exists` that tracks days when a frontmatter field exists and is non-empty. This is useful for habit tracking where you just need to know if something happened, regardless of the value.

# Changes
- Added `FrontmatterExists` to `SearchType` enum
- Implemented `collectDataFromFrontmatterExists()` function
- Added validation and parsing support for `"frontmatter.exists"`
- Updated all documentation (`docs/Concepts.md`, `docs/InputParameters.md`, `docs/TargetEvaluation.md`)
- Added comprehensive unit tests (12 test cases, all passing)
- Fixed pre-existing bug: `.contains()` → `.includes()` in `helper.ts`

# Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
- [x] Added unit tests (`test/frontmatter-exists.test.ts`) - all 12 tests passing
- [x] Added example file (`examples/TestFrontmatter.md`) demonstrating the feature
- [x] Tested backward compatibility - existing `frontmatter` behavior unchanged
- [x] Tested edge cases (empty strings, arrays, booleans, null, nested fields)
- [x] Tested with multiple frontmatter fields
- [x] Verified all tests run automatically without requiring Obsidian

# Documentation
- [x] Updated `docs/Concepts.md` - Added `frontmatter.exists` to searchType list
- [x] Updated `docs/InputParameters.md` - Added to searchType description
- [x] Updated `docs/TargetEvaluation.md` - Added detailed section for `frontmatter.exists`
- [x] Added `examples/TestFrontmatter.md` demonstrating the feature

# Related Issue
Closes #280

### CODE REVIEW

File: src/collecting.ts [hunk 1]
```diff
@@ -494,6 +494,44 @@ export function collectDataFromFrontmatterTag(
...
+    query: Query,
+    renderInfo: RenderInfo,
+    dataMap: DataMap,
+    xValueMap: XValueMap
+): boolean {
+    let frontMatter = fileCache.frontmatter;
+    if (frontMatter) {
+        let deepValue = helper.deepValue(frontMatter, query.getTarget());
+        // Check if field exists and is non-empty
+        if (deepValue !== null && deepValue !== undefined) {
```
Comment (lazyguru) [inline src/collecting.ts]: I'm a little rusty on JS recently, but I don't think it's possible for deepValue to be undefined here. The null check should be enough.

**Review by lazyguru — APPROVED** [review]
Code looks good. I'll test it out locally and then merge

**Comment by mikegreiner** [comment]
Interesting. I'll revisit the branch. Thanks for taking a look and sorry for the wasted time.

**Review by lazyguru — APPROVED** [review]

**Comment by lazyguru** [comment]
Included in 1.19.0
